adapters/huaweiads/huaweiads.go

```diff
@@ -701,34 +706,51 @@ func convertCountryCode(country string) (out string) {
 	return defaultCountryName
 }
 
+func getCountryCodeFromMCC(MCC string) (out string) {
+	var country_mcc = strings.Split(MCC, "-")[0]
+	intVar, _ := strconv.Atoi(country_mcc)
+
+	if result, found := MccList[intVar]; found {
+		return strings.ToUpper(result)
+	}
+	return defaultCountryName
+}
+
 // getDeviceID include oaid gaid imei. In prebid mobile, use TargetingParams.addUserData("imei", "imei-test");
 func getDeviceID(device *device, openRTBRequest *openrtb2.BidRequest) (err error) {
-	if openRTBRequest.User == nil {
-		return errors.New("getDeviceID: openRTBRequest.User is nil.")
-	}
-	if openRTBRequest.User.Ext == nil {
-		return errors.New("getDeviceID: openRTBRequest.User.Ext is nil.")
-	}
-	var extUserDataHuaweiAds openrtb_ext.ExtUserDataHuaweiAds
-	if err := json.Unmarshal(openRTBRequest.User.Ext, &extUserDataHuaweiAds); err != nil {
-		return errors.New("Unmarshal: openRTBRequest.User.Ext -> extUserDataHuaweiAds failed")
-	}
-	var deviceId = extUserDataHuaweiAds.Data
-	if len(deviceId.Imei) == 0 && len(deviceId.Gaid) == 0 && len(deviceId.Oaid) == 0 {
-		return errors.New("getDeviceID: Imei ,Oaid, Gaid are all empty.")
-	}
-	if len(deviceId.Oaid) > 0 {
-		device.Oaid = deviceId.Oaid[0]
-	}
-	if len(deviceId.Gaid) > 0 {
-		device.Gaid = deviceId.Gaid[0]
-	}
-	if len(deviceId.Imei) > 0 {
-		device.Imei = deviceId.Imei[0]
+	var userObjExist = true
+
+	if openRTBRequest.User == nil || openRTBRequest.User.Ext == nil {
+		userObjExist = false
 	}
-	if len(deviceId.ClientTime) > 0 {
-		device.ClientTime = getClientTime(deviceId.ClientTime[0])
+
+	if userObjExist {
+		var extUserDataHuaweiAds openrtb_ext.ExtUserDataHuaweiAds
+		if err := json.Unmarshal(openRTBRequest.User.Ext, &extUserDataHuaweiAds); err != nil {
+			return errors.New("Unmarshal: openRTBRequest.User.Ext -> extUserDataHuaweiAds failed")
+		}
+		var deviceId = extUserDataHuaweiAds.Data
+		if len(deviceId.Imei) == 0 && len(deviceId.Gaid) == 0 && len(device.Gaid) == 0 && len(deviceId.Oaid) == 0 {
+			return errors.New("getDeviceID: Imei ,Oaid, Gaid are all empty.")
+		}
+		if len(deviceId.Oaid) > 0 {
+			device.Oaid = deviceId.Oaid[0]
+		}
+		if len(deviceId.Gaid) > 0 {
+			device.Gaid = deviceId.Gaid[0]
+		}
+		if len(deviceId.Imei) > 0 {
+			device.Imei = deviceId.Imei[0]
+		}
+		if len(deviceId.ClientTime) > 0 {
+			device.ClientTime = getClientTime(deviceId.ClientTime[0])
+		}
+	} else {
+		if len(device.Gaid) == 0 {
```

Comment:
Please consider adding/updating existing JSON test cases to add coverage for this change